### ISSUE TITLE
Elastic Transcoder doesn't allow empty notifications.

### ISSUE THREAD

**Issue by tecnobrat** [hashicorp/terraform#8163]
This exists on terraform 0.7.0

Elastic Transcoder allows setting notifications to empty strings, however it appears that terraform flattens the list of notifications, removing any empty strings.  This is not the correct behaviour.  Infact, you should do the opposite, allow the user to not set the notification and set it as a blank string by default:

https://github.com/hashicorp/terraform/blob/b8a84691eceab2e6b70e2907ccb0aa06116b32e7/builtin/providers/aws/resource_aws_elastic_transcoder_pipeline.go#L251-L277

Currently, with this config:

```
    notifications = {
        completed   = "${aws_sns_topic.rise_transcode_sns.arn}"
        error       = "${aws_sns_topic.rise_transcode_sns.arn}"
        progressing = ""
        warning     = ""
    }
```

This error occurs:

```
* aws_elastictranscoder_pipeline.rise_transcoder_pipeline: Error creating Elastic Transcoder Pipeline: ValidationException: [The "Progressing" field is not allowed to be null., The "Warning" field is not allowed to be null.]
	status code: 400, request id: ee0d3d7c-60b9-11e6-836a-1da35ca08e88`
```

Some extra context:

https://github.com/boto/boto/issues/1315
https://github.com/aws/aws-sdk-net/issues/322

**Comment by tecnobrat** [comment]
Also appears after it failed to do the modification (the resource was previously created with the same arn for each notification type), it also saved the modified state file, even though it failed.  So now the state is inconsistent from the actual AWS settings

**Comment by catsby** [comment]
Thanks for writing in! This should be patched in #8207
